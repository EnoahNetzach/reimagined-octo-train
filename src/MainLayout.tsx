import { Outlet } from 'react-router-dom'
import Characters from './Characters.tsx'

export default function MainLayout() {
  return (
    <div className="p-4 h-screen grid gap-8 grid-cols-main grid-rows-1 text-white bg-black">
      <aside className="overflow-scroll">
        <Characters />
      </aside>

      <main className="max-w-3xl">
        <Outlet />
      </main>
    </div>
  )
}
