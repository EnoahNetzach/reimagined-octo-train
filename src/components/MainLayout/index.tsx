import { useEffect } from 'react'
import { Outlet, useLocation } from 'react-router-dom'
import Characters from '~/components/Characters'

export default function MainLayout() {
  const { key } = useLocation()

  useEffect(() => {
    window.scrollTo(0, 0)
  }, [key])

  return (
    <div className="p-4 h-screen grid gap-8 grid-cols-1 grid-rows-none md:grid-cols-main md:grid-rows-1 text-white bg-black">
      <aside className="order-2 md:order-1 md:overflow-scroll">
        <Characters />
      </aside>

      <main className="order-1 md:order-2 md:max-w-3xl md:overflow-scroll">
        <Outlet />
      </main>
    </div>
  )
}
