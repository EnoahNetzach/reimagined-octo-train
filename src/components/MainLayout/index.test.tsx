import { createRenderer } from 'react-test-renderer/shallow'
import MainLayout from '.'

jest.mock(
  '~/components/Characters',
  () =>
    function Characters() {
      return <></>
    },
)

describe('MainLayout', () => {
  it('mounts', () => {
    const renderer = createRenderer()
    renderer.render(<MainLayout />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })
})
