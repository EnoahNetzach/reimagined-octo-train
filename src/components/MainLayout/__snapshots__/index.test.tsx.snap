// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout mounts 1`] = `
<div
  className="p-4 h-screen grid gap-8 grid-cols-main grid-rows-1 text-white bg-black"
>
  <aside
    className="overflow-scroll"
  >
    <Characters />
  </aside>
  <main
    className="max-w-3xl overflow-scroll"
  >
    <Outlet />
  </main>
</div>
`;
